Add jms/serializer enum support

javer opened the pull request:
`jms/serializer` introduced enum support since [v3.20](https://github.com/schmittjoh/serializer/releases/tag/3.20.0) and added a new type: https://github.com/schmittjoh/serializer/blob/402a12629e3ad6d6897273769808a28c32745309/src/Metadata/Driver/EnumPropertiesDriver.php#L50

And since `jms/serializer-bundle` [v5.2.1](https://github.com/schmittjoh/JMSSerializerBundle/releases/tag/5.2.1) and https://github.com/schmittjoh/JMSSerializerBundle/pull/919 with `jms_serializer.enum_support: true` all existing models with enums got broken, because now enum properties are referenced to the fixed string `#/components/schemas/enum` instead of the correct `#/components/schemas/EnumClassName`:
```json
{
    "openapi": "3.0.0",
    "info": {},
    "paths": {
        "/api/card/random": {
            "get": {
                "operationId": "get_app_randomcardaction__invoke",
                "responses": {
                    "200": {
                        "description": "OK",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "$ref": "#/components/schemas/Card"
                                }
                            }
                        }
                    }
                }
            }
        }
    },
    "components": {
        "schemas": {
            "Card": {
                "properties": {
                    "rank": {
                        "type": "integer"
                    },
                    "suit": {
                        "$ref": "#/components/schemas/enum"
                    }
                },
                "type": "object"
            },
            "enum": {}
        }
    }
}
```

Before it was:
```json
{
    "openapi": "3.0.0",
    "info": {},
    "paths": {
        "/api/card/random": {
            "get": {
                "operationId": "get_app_randomcardaction__invoke",
                "responses": {
                    "200": {
                        "description": "OK",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "$ref": "#/components/schemas/Card"
                                }
                            }
                        }
                    }
                }
            }
        }
    },
    "components": {
        "schemas": {
            "Card": {
                "properties": {
                    "rank": {
                        "type": "integer"
                    },
                    "suit": {
                        "$ref": "#/components/schemas/Suit"
                    }
                },
                "type": "object"
            },
            "Suit": {
                "type": "string",
                "enum": [
                    "hearts",
                    "diamonds",
                    "clubs",
                    "spades"
                ]
            }
        }
    }
}
```

This PR adds support for enum type going from `jms/serializer` and returns the correct behavior.

theofidry commented:
I unfortunately have no permissions on this package, only the Alice one.

Yozhef commented:
@GuilhemN саn you review please?

javer commented:
Is there any update on this?

szepczynski commented:
@GuilhemN maybe you can help with review and merge this PR?

javer commented:
Any updates on this? @GuilhemN could you please take a look at this small fix?

DjordyKoert commented:
@javer can you please write test(s) for this new feature? Also please rebase on master 😄

javer commented:
@DjordyKoert Done: rebased on master, added test and even fixed codestyle in master 😄
Second commit is intended for keeping git history of JMSController.

DjordyKoert commented on ModelDescriber/JMSModelDescriber.php:
```diff
@@ -282,6 +282,15 @@ public function describeItem(array $type, OA\Schema $property, Context $context)
             $property->type = 'string';
             $property->format = 'date-time';
         } else {
+            // See https://github.com/schmittjoh/serializer/blob/master/src/Metadata/Driver/EnumPropertiesDriver.php#L51
```
Its better to make this a permalink

DjordyKoert commented on Tests/Functional/TestKernel.php:
```diff
@@ -310,6 +310,12 @@ protected function configureContainer(ContainerBuilder $c, LoaderInterface $load
             ],
         ]);
 
+        if ($this->flags & self::USE_JMS && \PHP_VERSION_ID >= 80100) {
+            $c->loadFromExtension('jms_serializer', [
+                'enum_support' => true,
+            ]);
+        }
```
`jms_serializer.enum_support` defaults to `false` even in PHP 8.1 or higher according to https://github.com/schmittjoh/JMSSerializerBundle/blob/master/Resources/doc/configuration.rst#extension-reference.
> `enum_support: true # PHP 8.1 Enums support, false by default for backward compatibility`

Might be better to test both with `false` & `true` options by adding an additional flag for this? Gonna leave this one for @GuilhemN to decide if this is needed or not.

javer replied:
With `enum_support: false` test passes even without my change in `JMSModelDescriber`, so it would be strange to write a test which is not failing without the fix.

DjordyKoert reviewed (approved):
LGTM

shakaran commented:
LGTM :+1:

shakaran commented:
@javer could you rebase the conflicts? Thanks

javer commented:
Done, rebased on the latest master, and all tests are green.

DjordyKoert commented:
Thank you @javer!
